algorithms/src/Kokkos_Random.hpp

```diff
@@ -59,1214 +59,1184 @@
-    in contrast to CuRand none of the functions of the pool (or the generator) are collectives,
-    i.e. all functions can be called inside conditionals.
+       //Max value returned by draw(Generator& gen)
+       KOKKOS_INLINE_FUNCTION
+       static Scalar max();
 
-    template<class Device>
-    class Pool {
-     public:
-      //The Kokkos device type
-      typedef Device device_type;
-      //The actual generator type
-      typedef Generator<Device> generator_type;
+       //Returns a value between zero and max()
+       KOKKOS_INLINE_FUNCTION
+       static Scalar draw(Generator& gen);
 
-      //Default constructor: does not initialize a pool
-      Pool();
+       //Returns a value between zero and range()
+       //Note: for floating point values range can be larger than max()
+       KOKKOS_INLINE_FUNCTION
+       static Scalar draw(Generator& gen, const Scalar& range){}
 
-      //Initializing constructor: calls init(seed,Device_Specific_Number);
-      Pool(unsigned int seed);
-
-      //Intialize Pool with seed as a starting seed with a pool_size of num_states
-      //The Random_XorShift64 generator is used in serial to initialize all states,
-      //thus the intialization process is platform independent and deterministic.
-      void init(unsigned int seed, int num_states);
-
-      //Get a generator. This will lock one of the states, guaranteeing that each thread
-      //will have its private generator. Note: on Cuda getting a state involves atomics,
-      //and is thus not deterministic!
-      generator_type get_state();
+       //Return value between start and end
+       KOKKOS_INLINE_FUNCTION
+       static Scalar draw(Generator& gen, const Scalar& start, const Scalar&
+  end);
+    };
 
-      //Give a state back to the pool. This unlocks the state, and writes the modified
-      //state of the generator back to the pool.
-      void free_state(generator_type gen);
+  The Random number generators themselves have two components a state-pool and
+  the actual generator A state-pool manages a number of generators, so that each
+  active thread is able to grep its own. This allows the generation of random
+  numbers which are independent between threads. Note that in contrast to CuRand
+  none of the functions of the pool (or the generator) are collectives, i.e. all
+  functions can be called inside conditionals.
+
+  template<class Device>
+  class Pool {
+   public:
+    //The Kokkos device type
+    typedef Device device_type;
+    //The actual generator type
+    typedef Generator<Device> generator_type;
+
+    //Default constructor: does not initialize a pool
+    Pool();
+
+    //Initializing constructor: calls init(seed,Device_Specific_Number);
+    Pool(unsigned int seed);
+
+    //Intialize Pool with seed as a starting seed with a pool_size of num_states
+    //The Random_XorShift64 generator is used in serial to initialize all
+  states,
```

Comment:
This should be on a commented line.  Same for the next few lines

Reply:
This is all in a big comment block basically just documentation

Reply:
I stand with Bruno on this one.  This formatting makes no sense.  Please fix or do `// clang-format off` before the comment block and turn back on after.

Reply:
I turned clang-format off before the comment block and on afterwards.